Bring back the sun indicator

MalteSchm opened the pull request:
@helgeerbe mentioned here: https://github.com/helgeerbe/OpenDTU-OnBattery/pull/172#issuecomment-1523468992 that I broke the sun indicator for Power Limiter solar pass-through. I investigated and believe the code in this PR fixes this. It works for me but I hardly ever get direct SolarPassthrough in my setup. 
Can you kindly check?

Note: I did not compare the MPPT power and the inverter power to indicate if the battery is actually being discharged or charged. Is that the desired behavior or shall this be done?
Note 2: The UI shows the last requested inverter power currently. This is correct as per methods used currently but also means that there is a power value shown case if the inverter is off. Should this indicate 0 in this case?

madmartin commented:
@MalteSchm  can you rebase your three commits into one and do a "git push -f" ?
In all your PRs there are unnecessary many commits that make the main history look messy.
If you need assistance let me know.

MalteSchm commented:
Hi @madmartin 

two questions: 
- The reason why there is a merge in this branch is that I made a commit followed by a push. Then I tried to amend that commit (to not create additional commits). This triggers this merge after which I need to commit again. So instead of doing one commit I end up doing three. Is there a suggested way to amend a commit that I've pushed to the origin without running into this issue? 
- When doing git rebase: What should be the new base? Is that "git rebase upstream/development" or "git rebase -i". The last time I tried this I ended up with multiple commits and it made things worse.
 
Sorry for the hassle. I have not really become accustomed to git yet.

MalteSchm commented:
thanks @madmartin 
I also saw that this has been merged but I walked through the steps and ended up with a single commit in my branch 
(thanks a lot ! for the detailed explanation)

... not sure where you got the lalala lines but I'd be _very_ surprised if I wrote them. 

@helgeerbe 
The code currently checked in development still checks for MPPT producing to detect a battery charging

```
    if(VeDirect.veFrame.PPV > 0) {
      return PL_UI_STATE_CHARGING;
    }
```

whereas you suggested above to check if the inverter is not producing:

```
     if(!inverter->isProducing()) {
       // battery charging icon
       return PL_UI_STATE_CHARGING;
     }
```

I changed this yesterday but could not commit as we had a network outage. I updated the branch just now. Let me know if this should be changed.

MalteSchm commented:
@madmartin 
I tried this on another branch https://github.com/MalteSchm/OpenDTU-OnBattery/tree/toggle_inverter_fix but this approach leads me into an infinite loop.
upon completion of git rebase (after selecting the commits, merging and doing the git rebase --continue(s)) I cannot commit the result 

```
$ git rebase --continue
Successfully rebased and updated refs/heads/toggle_inverter_fix.
$ git push
To https://github.com/MalteSchm/OpenDTU-OnBattery.git
 ! [rejected]        toggle_inverter_fix -> toggle_inverter_fix (non-fast-forward)
error: failed to push some refs to 'https://github.com/MalteSchm/OpenDTU-OnBattery.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

From there if I do a git pull I need to merge once more. This needs to be followed by rebase again to get rid of the additional commits. I keep arriving at this point with more and more commits in my branch.

Git and I are no real friends yet
